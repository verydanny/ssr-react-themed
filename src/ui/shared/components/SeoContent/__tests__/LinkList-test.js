import React from 'react'
import { render } from 'enzyme'
import toJson from 'enzyme-to-json'
import ThemedLinkList from '../LinkList'

const LinkList = ThemedLinkList.WrappedComponent

const locationName = 'Foo City'

const linkData = [{
  displayName: 'Studio Apartments',
  tagItem: 'studio',
  url: '/california/santa-monica-apartments/studio',
}, {
  displayName: '1 Bedroom Apartments',
  tagItem: '1_bedroom',
  url: '/california/santa-monica-apartments/1-bedroom',
}, {
  displayName: '2 Bedroom Apartments',
  tagItem: '2_bedroom',
  url: '/california/santa-monica-apartments/2-bedroom',
}, {
  displayName: '3 Bedroom Apartments',
  tagItem: '3_bedroom',
  url: '/california/santa-monica-apartments/3-bedroom',
}, {
  displayName: 'Pet Friendly Apartments',
  tagItem: 'pet_friendly',
  url: '/california/santa-monica-apartments/pet-friendly',
}, {
  displayName: 'Furnished Apartments',
  tagItem: 'furnished',
  url: '/california/santa-monica-apartments/furnished',
}, {
  displayName: 'Cheap Apartments',
  tagItem: 'cheap',
  url: '/california/santa-monica-apartments/cheap',
}]

describe('ui/shared/components/SeoContent/LocalInfo', () => {
  it('renders the correct output for 7 items', () => {
    const wrapper = render(<LinkList links={linkData} header={`${locationName} Apartment Options`} />)

    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders the correct output for 7 items, with only one item being hidden (does not collapse)', () => {
    const wrapper = render(<LinkList links={linkData} maxLinksShowing={6} header={`${locationName} Apartment Options`} />)

    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
