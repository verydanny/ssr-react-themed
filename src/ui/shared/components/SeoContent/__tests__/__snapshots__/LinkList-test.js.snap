// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui/shared/components/SeoContent/LocalInfo renders the correct output for 7 items 1`] = `
<div
  data-tag_section="local_info"
  data-tid="link-list"
>
  <div
    data-tid="collapsible-section"
  >
    <h2
      data-tag_item="header"
    >
      <button>
        Foo City Apartment Options
      </button>
    </h2>
    <div>
      <div
        class=""
      >
        <ul
          class=""
        >
          <li
            class=""
            data-tid="list-item-0"
            index="0"
          >
            <a
              data-tag_item="studio"
              href="/california/santa-monica-apartments/studio"
            >
              Studio Apartments
            </a>
          </li>
          <li
            class=""
            data-tid="list-item-1"
            index="1"
          >
            <a
              data-tag_item="1_bedroom"
              href="/california/santa-monica-apartments/1-bedroom"
            >
              1 Bedroom Apartments
            </a>
          </li>
          <li
            class=""
            data-tid="list-item-2"
            index="2"
          >
            <a
              data-tag_item="2_bedroom"
              href="/california/santa-monica-apartments/2-bedroom"
            >
              2 Bedroom Apartments
            </a>
          </li>
          <li
            class=""
            data-tid="list-item-3"
            index="3"
          >
            <a
              data-tag_item="3_bedroom"
              href="/california/santa-monica-apartments/3-bedroom"
            >
              3 Bedroom Apartments
            </a>
          </li>
          <li
            class=""
            data-tid="list-item-4"
            index="4"
          >
            <a
              data-tag_item="pet_friendly"
              href="/california/santa-monica-apartments/pet-friendly"
            >
              Pet Friendly Apartments
            </a>
          </li>
        </ul>
        <div
          class=""
        >
          <ul
            class=""
          >
            <li
              class=""
              data-tid="list-item-0"
              index="0"
            >
              <a
                data-tag_item="furnished"
                href="/california/santa-monica-apartments/furnished"
              >
                Furnished Apartments
              </a>
            </li>
            <li
              class=""
              data-tid="list-item-1"
              index="1"
            >
              <a
                data-tag_item="cheap"
                href="/california/santa-monica-apartments/cheap"
              >
                Cheap Apartments
              </a>
            </li>
          </ul>
        </div>
        <button
          class=""
          data-tid="button"
        >
          <span>
            More
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`ui/shared/components/SeoContent/LocalInfo renders the correct output for 7 items, with only one item being hidden (does not collapse) 1`] = `
<div
  data-tag_section="local_info"
  data-tid="link-list"
>
  <div
    data-tid="collapsible-section"
  >
    <h2
      data-tag_item="header"
    >
      <button>
        Foo City Apartment Options
      </button>
    </h2>
    <div>
      <div
        class=""
      >
        <ul
          class=""
        >
          <li
            class=""
            data-tid="list-item-0"
            index="0"
          >
            <a
              data-tag_item="studio"
              href="/california/santa-monica-apartments/studio"
            >
              Studio Apartments
            </a>
          </li>
          <li
            class=""
            data-tid="list-item-1"
            index="1"
          >
            <a
              data-tag_item="1_bedroom"
              href="/california/santa-monica-apartments/1-bedroom"
            >
              1 Bedroom Apartments
            </a>
          </li>
          <li
            class=""
            data-tid="list-item-2"
            index="2"
          >
            <a
              data-tag_item="2_bedroom"
              href="/california/santa-monica-apartments/2-bedroom"
            >
              2 Bedroom Apartments
            </a>
          </li>
          <li
            class=""
            data-tid="list-item-3"
            index="3"
          >
            <a
              data-tag_item="3_bedroom"
              href="/california/santa-monica-apartments/3-bedroom"
            >
              3 Bedroom Apartments
            </a>
          </li>
          <li
            class=""
            data-tid="list-item-4"
            index="4"
          >
            <a
              data-tag_item="pet_friendly"
              href="/california/santa-monica-apartments/pet-friendly"
            >
              Pet Friendly Apartments
            </a>
          </li>
          <li
            class=""
            data-tid="list-item-5"
            index="5"
          >
            <a
              data-tag_item="furnished"
              href="/california/santa-monica-apartments/furnished"
            >
              Furnished Apartments
            </a>
          </li>
          <li
            class=""
            data-tid="list-item-6"
            index="6"
          >
            <a
              data-tag_item="cheap"
              href="/california/santa-monica-apartments/cheap"
            >
              Cheap Apartments
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
